LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -197,58 +271,113 @@ List<Module> GetModuleList()
+            ReportHeader(actionName: action);
+
+            if (ClassroomList.Count == 0)
             {
-                ReportError("Instructor", "Classroom");
+                ReportError(missingResource: "Instructor", failedAction: action);
             }
             else
             {
-                foreach (var classroom in classroomList)
+                foreach (var classroom in ClassroomList)
                 {
-                    if (moduleInstructorDictionary.ContainsKey(classroom.Module.Id))
+                    if (ModuleInstructorDictionary.ContainsKey(classroom.Module.Id))
                     {
                         List<Instructor> instructorList;
-                        moduleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
+                        ModuleInstructorDictionary.TryGetValue(classroom.Module.Id, out instructorList);
                         classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
                     }
                     else
                     {
-                        ReportError("Instructor", "Classroom");
+                        ReportError(missingResource: "Instructor", failedAction: action);
                     }
                 }
             }
+
+            ReportSummary(actionName: action, itemCount: 1);
+            ReportFooter(actionName: action);
         }
 
         public void RegisterStudentsForClasses()
         {
-           
+            string action = "Register Students For Classes";
+            int classroomItemListCount = 0;
+            ReportHeader(actionName: action);
+
+            if (ClassroomList.Count == 0)
+            {
+                ReportError(missingResource: "Instructor", failedAction: action);
+            }
+            else
+            {
+                foreach (var classroom in ClassroomList)
+                {
+                    if(classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        classroomItemListCount = classroom.ItemList.Count;
+                    } 
+                   
+                }
+            }
+
+            ReportSummary(actionName: action, itemCount: classroomItemListCount);
+            ReportFooter(actionName: action);
+        }
+
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            int itemListCount = AttributeGenerator.random.Next(0, classroom.MaximumCapacity - classroom.ItemList.Count + 1);
+            int i = 0;
+            while (i < itemListCount)
+            {
+                foreach (var student in ActiveStudentList)
+                {
+                    if (student.CompletedModuleList.Contains(classroom.Module))
+                    {
+                        classroom.ItemList.Add(student);
+
+                        ReportItem(itemName: student.ToString(), actionName: "", itemIndex: i);
```

Comment:
Don't use empty actionName, you can use "Registered for class" or something similar.